GitCommands/CommitMessageManager.cs

```diff
@@ -227,14 +201,44 @@ internal static string FormatCommitMessage(string commitMessage, bool usingCommi
 
         private string GetFilePath(string workingDirGitDir, string fileName) => _fileSystem.Path.Combine(workingDirGitDir, fileName);
 
-        private (string filePath, bool fileExists) GetMergeOrCommitMessagePath()
+        private string GetMergeOrCommitMessagePath() => IsMergeCommit ? MergeMessagePath : CommitMessagePath;
+
+        private string ReadFile(string filePath, string errorTitle, Encoding? encoding = null)
+        {
+            try
+            {
+                if (!_fileSystem.File.Exists(filePath))
+                {
+                    return string.Empty;
+                }
+
+                return _fileSystem.File.ReadAllText(filePath, encoding ?? Encoding.Default);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(null, string.Format(CannotAccessFile, ex.Message, filePath), errorTitle,
```

Comment:
Let it bubble up to the global handler?

Reply:
No, I agree to what you wrote "in this specific scenario we need to ignore the exception".
Just let the user know that something went wrong but I see no need to cancel the other operations in FormCommit.

Reply:
(If these file accesses would fail often, a retry should be offered here.)

Reply:
I think we should leave a comment here explaining why we're swallowing the exception and not sending to the global handler.